core/src/main/java/io/github/tcdl/collector/Collector.java

```diff
@@ -49,23 +46,30 @@
     private TimeoutManager timeoutManager;
 
     private Clock clock;
-
+    private MsbContextImpl msbContext;
```

Comment:
I'm not sure that put msbContext inside Collector is good idea,as I understand MsbContext holds everything for creating Requester,ResponderServer etc..So with MsbContext  we create  Requester and it holds Collector with MsbContext as a field,very complex)
Also "private MsbContextImpl msbContext;"=>"private MsbContext msbContext;"